sklearn/compose/tests/test_column_transformer.py

```diff
@@ -368,6 +368,37 @@ def test_column_transformer_sparse_stacking():
     assert_array_equal(X_trans[:, 1:], np.eye(X_trans.shape[0]))
 
 
+def test_column_transformer_mixed_cols_sparse():
+    pd = pytest.importorskip('pandas')
+    df = pd.DataFrame([['a', 1, True],
+                       ['b', 2, False]],
+                      columns=['categorical', 'numerical', 'boolean'])
+
+    ct = make_column_transformer(
+        (['categorical'], OneHotEncoder()),
+        (['numerical', 'boolean'], 'passthrough'),
+        sparse_threshold=1.0
+    )
+
+    # this shouldn't fail, since boolean can be coerced into a numeric
+    # See: https://github.com/scikit-learn/scikit-learn/issues/11912
+    X_trans = ct.fit_transform(df)
+    assert X_trans.getformat() == 'csr'
+    assert_array_equal(X_trans.toarray(), np.array([[1, 0, 1, 1],
+                                                    [0, 1, 2, 0]]))
+
+    ct = make_column_transformer(
+        (['categorical'], OneHotEncoder()),
+        (['categorical'], 'passthrough'),
+        sparse_threshold=1.0
+    )
+    with pytest.raises(ValueError,
+                       match="For a sparse output, all columns should"):
+        # this fails since strings `a` and `b` cannot be
```

Comment:
Do we not need similar behaviour for the dense case?

Reply:
Not necessarily, as numpy can concatenate those. 
Although it will not necessarily do what you want (eg convert all numbers to strings, if one of them the subarrays is of string dtype), but for the specific case of the example here with object boolean values, it will be able to concatenate and return an object array, which will be converted to a numeric array in a consequent sklearn estimator.